roles/network/tasks/hostapd.yml

```
@@ -1,7 +1,17 @@
 - name: Turn off hostapd when no wifi interface present or in "Appliance Mode"
   set_fact:
     hostapd_enabled: False
-  when: not iiab_wireless_lan_iface is defined or iiab_network_mode == "Appliance"
+  when: iiab_wireless_lan_iface is undefined or iiab_network_mode == "Appliance"
+
+- name: Detect current Wifi channel
+  shell: iw {{ discovered_wireless_iface }} info | grep channel | cut -d' ' -f2
+  register: current_host_channel
+  when: discovered_wireless_iface != "none"
+
+- name: Setting WiFi channel to {{ current_host_channel.stdout }} on RPi hardware
+  set_fact:
+    host_channel: "{{ current_host_channel.stdout }}"
+  when: current_host_channel.stdout != "" and current_host_channel.stdout|int <= 13 and discovered_wireless_iface != "none" and rpi_model != "none"
```

I don't know if this is what's needed, but the following allows installation to continue on a Debian 10.3 VM:

```suggestion
  when: current_host_channel.stdout is defined and current_host_channel.stdout != "" and current_host_channel.stdout|int <= 13 and discovered_wireless_iface != "none" and rpi_model != "none"
```

---

that works